src/IMG_anim_decoder.c

```diff
@@ -193,14 +193,21 @@ bool IMG_GetAnimationDecoderFrame(IMG_AnimationDecoder *decoder, SDL_Surface **f
     if (!duration) {
         duration = &temp_duration;
     }
-
+    
     SDL_ClearError();
+    // Reset the status before trying to get the next frame.
+    decoder->status = IMG_CODER_STATUS_OK;
 
     bool result = decoder->GetNextFrame(decoder, frame, duration);
     if (temp_frame) {
         SDL_DestroySurface(temp_frame);
     }
 
+    // If the underlying decoder only uses the SDL_SetError function, we can update our status accordingly as well.
+    if (decoder->status == IMG_CODER_STATUS_OK && SDL_GetError()[0] != '\0') {
+        decoder->status = IMG_CODER_STATUS_FAILED;
+    }
```

Comment:
```suggestion
    // If the underlying decoder only uses the SDL_SetError function, we can update our status accordingly as well.
    if (!result) {
        if (SDL_GetError()[0] == '\0') {
            decoder->status = IMG_CODER_STATUS_COMPLETE;
        } else {
            decoder->status = IMG_CODER_STATUS_FAILED;
        }
    }
```

Reply:
SDL functions can set the error internally and then recover successfully, leaving the error still set.